spec/duration.html

```diff
@@ -1511,7 +1511,7 @@ <h1>TemporalDurationToString ( _years_, _months_, _weeks_, _days_, _hours_, _min
           1. If _precision_ is *"auto"*, then
             1. If _fraction_ ≠ 0, then
               1. Let _decimalPart_ be _fraction_ formatted as a nine-digit decimal number, padded to the left with zeroes if necessary.
-              1. Set _fractionString_ to the longest possible substring of _decimalPart_ starting at position 0 and not ending with the code unit 0x0030 (DIGIT ZERO).
+              1. Let _fractionString_ be the longest possible substring of _decimalPart_ starting at position 0 and not ending with the code unit 0x0030 (DIGIT ZERO).
```

Comment:
Hah, the linter I wrote did not understand that this definition doesn't apply to the use 4 lines down (it's just a scope walk, no control flow). Oops, there's still a bug here, but fixing in #1405 .

Reply:
ecmarkup has a linter built in; it might be nice to improve that (or replace it with your linter, if that’s a better direction). cc @bakkot

Reply:
This proposal is actually [using the linter](https://github.com/tc39/proposal-temporal/blob/49629f785eee61e9f6641452e01e995f846da3a1/package.json#L33) already!

Re: scope analysis, I've started down that path, but there's [a whole bunch of special cases](https://github.com/tc39/ecmarkup/issues/173#issuecomment-626272511) which I've been slowly [cleaning up](https://github.com/tc39/ecma262/pull/2258). PRs welcome!

Reply:
I handled all of those cases I found in the temporal proposal, which is to say fewer than exist in ecma262, but my current linter is a messy python script thrown together as I encountered things - you probably don't want it in its current state. It was designed as a one-off because I saw recurring issues from what I presume were rebases + copy&paste.

That being said, when I get some extra time I'd love to upstream more of its functionality to ecmarkup; mind giving me some code pointers (IRC?) when I get around to it? I went with the quick python hack because poking around the ecmarkup code left me with some questions.

Reply:
Sure! I'm bakkot on IRC in #tc39 and #tc39-delegates; feel free to ping me whenever.